src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -201,6 +201,51 @@ public override AssemblyIdentity Identity
             }
         }
 
+        internal override bool HasGuidAttribute
+        {
+            get
+            {
+                foreach (var attrData in GetAttributes())
+                {
+                    if (attrData.IsTargetAttribute(this, AttributeDescription.GuidAttribute))
```

Comment:
It looks like we validate this attribute in ```DecodeWellKnownAttribute```, but don't record its information in ```CommonAssemblyWellKnownAttributeData```. Perhaps we should do just that.